mvpinternshipapp.client/src/components/sale/SaleDetails.jsx

```diff
@@ -0,0 +1,111 @@
+import { Input, Label, Dropdown } from "semantic-ui-react";
+import { useState, useEffect } from 'react';
+import api from '../../api';
+
+import PropTypes from 'prop-types';
+
+export function SaleDetails({ state, setState, open }) {
+    const [customers, setCustomers] = useState([]);
+    const [products, setProducts] = useState([]);
+    const [stores, setStores] = useState([]);
+
+    useEffect(() => {
+        if (open) {
+            let stillAlive = true;
+            const mapToOptions = (objArray) => {
+                return objArray.map(obj => ({
+                    key: obj.id,
+                    value: obj.id,
+                    text: obj.name,
+                }));
+            };
+            api.getCustomers().then((res) => {
```

Comment:
You already have .then, where is **catch** to handle errors?

Can you please check and implement it to all of your pages.